webpack/assets/javascripts/react_app/components/common/ErrorBoundary/index.js

```diff
@@ -41,15 +42,23 @@ class ErrorBoundary extends React.Component {
         <p>
           {__('There was a problem processing the request. Please try again.')}
         </p>
-        <p
-          dangerouslySetInnerHTML={{
-            __html: __(
-              `To report an issue <a target="_blank" rel="noopener noreferrer" href=${foremanUrl(
-                '/links/issues'
-              )}>click here</a>`
-            ),
-          }}
-        />
+        <p>
+          <FormattedMessage
+            id="report-issue"
+            defaultMessage={__('To report an issue {click}')}
+            values={{
+              click: (
+                <a
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  href={foremanUrl('/links/issues')}
+                >
+                  click here
```

Comment:
This is now untranslated. You can try to use `{link_start}` and `{link_end}` as placeholders. So the formatted message becomes `To report an issue {link_start}click here{link_end}`. That allows translators to create a nice sentence while still having a link.

Reply:
![Screenshot from 2021-11-18 11-24-55](https://user-images.githubusercontent.com/30431079/142399066-13e9569b-5fa3-4191-b957-8d242e32fcd2.png)
I see it as translated, and I don't think we can use `{link_start}` and `{link_end}` as `values` values are being resolved

Reply:
I meant that you can provide those as context instead of providing `click` as a context.

Reply:
What do you mean by "context" here?

Reply:
Looks like it's called `values` here.

Reply:
And you mean `link_end : </a>` ? or something else?

Reply:
Exactly that.

Reply:
Thats not a valid XML tag so it isn't resolved and I get this:
![Screenshot from 2021-11-18 12-05-33](https://user-images.githubusercontent.com/30431079/142404001-722b78fa-9dbe-4974-a0f7-06c96b8e337a.png)

Reply:
That sounds like from a translators perspective it would be better to continue using `dangerouslySetInnerHTML` here.

A quick search directed me to https://reactjsexample.com/enhance-formatted-messages-for-react-intl/ which eventually led me to https://formatjs.io/docs/react-intl/components/#rich-text-formatting. That looks like the right path to separate the actual HTML from the translated message.

Reply:
I don't see how the link you sent is different from what this pr is doing?

Reply:
It looks to me you could use (using https://formatjs.io/docs/react-intl/components/#function-as-the-child):
```
<FormattedMessage
  id="report-issue"
  defaultMessage={__('To report an issue <a>click here</a>')}
  values={{
    a: text => <a target="_blank" rel="noopener noreferrer" href={foremanUrl('/links/issues')}>{text}</a>,
  }}
/>
```

Reply:
We are using `"react-intl": "^2.8.0"` and it looks like using a function as a child is only introduced in 3.0
https://formatjs.io/docs/react-intl/upgrade-guide-3x#enhanced-formattedmessage--formatmessage-rich-text-formatting

Reply:
I still think this is an issue and you'll get poorly translated sentences (both in `ConfigReports` and `ErrorBoundary`). Can't we update to 3.0?

Having said that, the current sentences will also be very bad (they use string interpolation which is always a bad sign) so I don't object to merging per se. That does mean we should have an issue tracking this and possibly even link to it in the code as a TODO.

Reply:
The update is being worked on, starting here: https://github.com/theforeman/foreman/pull/7910

Reply:
Should we wait on that or would you prefer moving this forward?

Reply:
I don't mind either way. And anyway the translations won't be extracted until https://github.com/theforeman/foreman/pull/8965 is in

Reply:
Then I'd prefer to hold this a bit and prioritize merging those two if we can.